.github/workflows/publish_release.yml

```diff
@@ -8,41 +8,40 @@ on:
-        uses: rohith/publish-nuget@v2
-        with:
-          PROJECT_FILE_PATH: src/FSharp.Data.GraphQL.Client/FSharp.Data.GraphQL.Client.fsproj # Relative to repository root
-          VERSION_FILE_PATH: Directory.Build.targets
-          VERSION_REGEX: <Version>(.*)<\/Version>
-          # VERSION_FILE_PATH: Directory.Build.props # Filepath with version info, relative to repository root. Defaults to project file
-          # VERSION_REGEX: <Version>(.*)<\/Version> # Regex pattern to extract version info in a capturing group
-          # TAG_COMMIT: true # Flag to enable / disalge git tagging
-          # TAG_FORMAT: v* # Format of the git tag, [*] gets replaced with version
-          NUGET_KEY: ${{secrets.NUGET_SECRET}}
-      - name: publish FSharp.Data.GraphQL.Server on version change
-        uses: rohith/publish-nuget@v2
-        with:
-          PROJECT_FILE_PATH: src/FSharp.Data.GraphQL.Server/FSharp.Data.GraphQL.Server.fsproj
-          VERSION_FILE_PATH: Directory.Build.targets
-          VERSION_REGEX: <Version>(.*)<\/Version>
-          NUGET_KEY: ${{secrets.NUGET_SECRET}}
-      - name: publish FSharp.Data.GraphQL.Server.Middleware on version change
-        uses: rohith/publish-nuget@v2
-        with:
-          PROJECT_FILE_PATH: src/FSharp.Data.GraphQL.Server.Middleware/FSharp.Data.GraphQL.Server.Middleware.fsproj
-          VERSION_FILE_PATH: Directory.Build.targets
-          VERSION_REGEX: <Version>(.*)<\/Version>
-          NUGET_KEY: ${{secrets.NUGET_SECRET}}
+    - uses: actions/checkout@v2
+    - name: Setup .NET Core
+      uses: actions/setup-dotnet@v1
+
+    - name: Pack FSharp.Data.GraphQL.Shared project and push to NuGet
+      if: startsWith(github.ref, 'refs/tags/release')
+      run: |
+        cd src/FSharp.Data.GraphQL.Shared
+        dotnet pack --nologo --configuration Release /p:IsNuget=true -o ../../nuget
+        dotnet nuget push ../../nuget/*Shared*.nupkg -k $KEY
+
+    - name: Pack FSharp.Data.GraphQL.Client project and push to NuGet
+      if: startsWith(github.ref, 'refs/tags/release')
+      run: |
+        cd src/FSharp.Data.GraphQL.Client
+        dotnet pack --nologo --configuration Release /p:IsNuget=true -o ../../nuget
+        dotnet nuget push ../../nuget/*Client*.nupkg -k $KEY
+
+    - name: Pack FSharp.Data.GraphQL.Server project and push to NuGet
+      if: startsWith(github.ref, 'refs/tags/release')
+      run: |
+        cd src/FSharp.Data.GraphQL.Server
+        dotnet pack --nologo --configuration Release /p:IsNuget=true -o ../../nuget
+        dotnet nuget push ../../nuget/*Server*.nupkg -k $KEY
+
+    - name: Pack FSharp.Data.GraphQL.Server.Middleware project and push to NuGet
+      if: startsWith(github.ref, 'refs/tags/release')
+      run: |
+        cd src/FSharp.Data.GraphQL.Server.Middleware
+        dotnet pack --nologo --configuration Release /p:IsNuget=true -o ../../nuget
+        dotnet nuget push ../../nuget/*Server.Middleware*.nupkg -k $KEY
```

Comment:
It's typically good to build the project in a separate job, save it as an artifact, test that artifact, and publish the same tested artifact only after that. All in the same workflow. To reduce duplication, reusable workflows can be used, they good enough these days.

Reply:
Could you show an example of that?

Reply:
I only have examples from the Python ecosystem, they will probably not help you. But it's easy. Use the `needs` attribute to chain stages and use `actions/upload-artifact` in the build job and `actions/download-artifact` in the test and publish ones.
I don't know anything about the dotnet ecosystem to suggest how to actually invoke testing — you'll need to figure it out yourself.

Reply:
@webknjaz Hello. Could you, please, send me examples from the Python ecosystem?